test/test_cpp_extensions_open_device_registration.py

```diff
@@ -333,29 +333,29 @@ def test_open_device_storage_pin_memory(self):
         # Check if the pin_memory is functioning properly on custom device
         cpu_tensor = torch.empty(3)
         self.assertFalse(cpu_tensor.is_foo)
-        self.assertFalse(cpu_tensor.is_pinned("foo"))
```

Comment:
I guess we're missing some privateuse1 initialization here?

Reply:
Actually not, it has been done in `setUpClass()` method. The error is first caused by [L347](https://github.com/wizzniu/pytorch/blob/update_pin_memory_api/test/test_cpp_extensions_open_device_registration.py#L347) and failed again at [L336](https://github.com/wizzniu/pytorch/blob/update_pin_memory_api/test/test_cpp_extensions_open_device_registration.py#L336) when retrying to run this case. We restore the default argument `device` of `storage.is_pinned/pin_memory` to 'cuda', so have to add "foo" argument back here.
As for another failed case in `TestDataLoader`, it's because cuda (and other backends) doesn't support `setCurrentDevice()` now when calling `torch._C._accelerator_hooks_set_current_device(device_id)`.We must get  https://github.com/pytorch/pytorch/pull/131854 merged first, which blocks this PR.

Reply:
@albanD Have a look at PR https://github.com/pytorch/pytorch/pull/131854 first?